### PR TITLE
Update the version of snakeyaml to 2.0 and changes to the code to mak…

### PR DESCRIPTION
**Pull request by sachinmaheshwari**
…e it compatible.

# Pull Request Template

## Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change
- Upgrade the version of snakeyaml version to 2.0
- Fix the compilation issues due to this version upgrade.
- Some minor refactoring that came in the static analysis.

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

**Test Configuration**:
* Firmware version:
* Hardware:
* Toolchain:
* SDK:

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have checked my code and corrected any misspellings

### CODE REVIEW

File: src/main/java/com/intuit/apl/engine/ParsingRules.java [hunk 1]
```diff
@@ -275,37 +279,13 @@ private void loadRuleFiles(String[] fileNames, InputStream inputStream, RulePack
...
-  // this.getClass().getClassLoader());
-  // parser = new SpelExpressionParser(config);
-  // }
-  // loadRuleFiles(fileNames, inputStream, rulePackage, yaml);
-  // }
-  //
   void parseRuleFiles(String[] fileNames, InputStream inputStream,
       PolicyRepository policyRepository) {
-    Yaml yaml = new Yaml(new Constructor(HashMap.class));
+    Yaml yaml = new Yaml();
```
Comment (ravichauhan03) [inline src/main/java/com/intuit/apl/engine/ParsingRules.java:284]: Do you wish to add the taginspector for not parsing malicious content ?

**Review by ravichauhan03 — COMMENTED** [review]
Please see the comments.

File: src/main/java/com/intuit/apl/engine/ParsingRules.java (hunk 1 above)
Comment (sachinmaheshwari) [reply src/main/java/com/intuit/apl/engine/ParsingRules.java:284]: By default, it doesn’t allow global tags to set thereby fixing that security vulnerability. This [article](https://www.veracode.com/blog/research/resolving-cve-2022-1471-snakeyaml-20-release-0) also talks about it.

**Review by ravichauhan03 — APPROVED** [review]
Approved.

**Pull request merged**